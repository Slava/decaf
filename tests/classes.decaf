class Animal {
  string name;
  void say () {
    Print("*make noise*");
  }
}

class Cat extends Animal implements Adorable {
  void say () {
    Print("meow");
  }
}

class Dog extends Animal implements Adorable, Protector {
  void say () {
    Print("gaw");
  }
}

