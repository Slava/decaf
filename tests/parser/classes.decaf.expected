Program
  Class Animal []
    <string name>
    Function void: say:
      (#Print "*make noise*")
  Class Cat: Animal [Adorable]

    Function void: say:
      (#Print "meow")
  Class Dog: Animal [Adorable Protector]

    Function void: say:
      (#Print "gaw")
